substrate/bin/node/cli/src/chain_spec.rs

```diff
@@ -353,7 +383,54 @@ fn configure_accounts(
 
 	let num_endowed_accounts = endowed_accounts.len();
 
-	(initial_authorities, endowed_accounts, num_endowed_accounts, stakers)
+	let genesis_shares = etf_genesis(
+		initial_authorities.iter().map(|x| x.7.clone()).collect::<Vec<_>>(),
+		vec!["Alice", "Bob"],
+	);
+	(initial_authorities, endowed_accounts, num_endowed_accounts, stakers, genesis_shares)
+}
+
+/// Helper function to prepare initial secrets and resharing for ETF conensus
+/// return a vec of (authority id, resharing, pubkey commitment)
+pub fn etf_genesis(
+	initial_authorities: Vec<BeefyId>, 
+	seeds: Vec<&str>
+) -> Vec<(BeefyId, BeefyId, Vec<u8>)> {
+	let msk = ark_bls12_377::Fr::rand(&mut OsRng);
+	let msk_prime = ark_bls12_377::Fr::rand(&mut OsRng);
+
+	let genesis_resharing = HighThresholdACSS::reshare(
+		msk, 
+		msk_prime, 
+		&initial_authorities.iter().map(|authority| {
+			// NO: that's 144 bytes, we only want the first 48 of them (48 + 96 bytes for both keypairs)
+			ark_bls12_377::G1Projective::deserialize_compressed(
+				&authority.to_raw_vec()[..48]
+			).unwrap()
+		}).collect::<Vec<_>>(), 
+		initial_authorities.len() as u8, 
+		&mut OsRng,
+	);
+
+	initial_authorities.iter().enumerate().map(|(idx, auth)| {
+		let pok = &genesis_resharing[idx].1;
+		let mut bytes = Vec::new();
+		pok.serialize_compressed(&mut bytes).unwrap();
```

Comment:
Do we need to unwrap this?

Reply:
We do. It could be an expect instead, but it's fine if this panics since it's a critical failure if the pok can't be serialized on genesis anyway.